server/hwapi/endpoints/certification/logic.py

```diff
@@ -31,30 +31,29 @@
 
 def find_main_hardware_components(
     db: Session, board_data: BoardValidator, bios_data: BiosValidator | None
-) -> tuple[models.Device, models.Bios | None]:
+) -> tuple[models.Device, Sequence[models.Bios]]:
```

Comment:
- why not separate this into two functions, one to find the board and another for bios?
- it's encouraged to be generic with what you accept and specific with what you return. In other words while it makes sense to take as input an abstract type like a Sequence, we should return a concrete type like a list. This is because the user of the function should know exactly what kind of object they can expect returned.